### PR TITLE
Import export in CLI

### PR DESCRIPTION
**Pull request by hhsecond**
#### _Why is this change required? What problem does it solve?:_
Introducing import-export utility over the command line

#### _If it fixes an open issue, please link to the issue here:_
This PR won't solve issue #72 completely but starting with image import and export. Also, the release is going to be experimental and the APIs are subjective to change

#### _Describe your changes in detail:_
- Moving CLI as a module
- Introducing import option using `click`
- Introducing export option using `click`
- Base class for the introduction of plugin system

## Types of changes
What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [x] Documentation update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Is this PR ready for review, or a work in progress?
- [ ] Ready for review
- [x] Work in progress

## How Has This Been Tested?
Put an `x` in the boxes that apply:
- [ ] Current tests cover modifications made
- [x] New tests have been added to the test suite
- [ ] Modifications were made to existing tests to support these changes
- [ ] Tests may be needed, but they are not included when the PR was proposed
- [ ] I don't know. Help!

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.rst)** document.
- [x] I have signed (or will sign when prompted) the tensorwork CLA.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rlizzo** [comment]
Hey @hhsecond I took the code you started on and did a massive overhaul of it. Basically took heavy inspiration from scikit-image's plugin infrastructure and utility. The interface needs to be cleaned up a bit still, and we need tests, but both importing and exporting works! 

Let me know what you think!

**Comment by hhsecond** [comment]
@rlizzo Wow! that's indeed massive. Are we pushing this in 0.2? I was thinking of a basic stable interface which we could push into 0.2

**Comment by rlizzo** [comment]
Why don't we push it to 0.3? We can make that a quick release.

 If we're going to put it in 0.2 I'd rather not include  a plugin the system (which we haven't had time to  thoroughly review and test.) I suppose we could just hard code pillow is a dependency and only allow the file types that it could support. That would serve as a decent proof-of-concept?

What are your thoughts? Should we put a limited version in 0.2? Or just push it all to 0.3? (It'll be a short turn around time for that release)

**Comment by hhsecond** [comment]
@rlizzo do you wanna go back to my last commit and push that for 0.2
It has exactly this. Only pillow loader nothing else. Then we can keep your updates for 0.3?

**Comment by rlizzo** [comment]
Hey @hhsecond can you give this a review? The tests are passing now (even though a few more are needed) and the quality is close to where I'm OK with it for the time being.

File: src/hangar/cli/io/manage_plugins.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+        for p in provides:
+            if p not in plugin_store:
+                print(f"Plugin: {name} wants to provide non-existent: {p}. Ignoring.")
+
+        plugin_provides[name] = valid_provides
+        plugin_module_name[name] = os.path.basename(filename)[:-4]
+
+
+_scan_plugins()
```
Comment (hhsecond) [inline src/hangar/cli/io/manage_plugins.py]: would be great (for readability) if all the function calls are placed together at one place. Or may be we could do it when we clean it up as a proper plugin system

File: src/hangar/cli/cli.py [hunk 2]
```diff
@@ -336,6 +334,133 @@ def server(overwrite, ip, port, timeout):
...
+
+        if os.path.isfile(path):
+            fname = os.path.basename(path)
+            if not overwrite:
+                if fname in aset:
+                    return None
+            fNamePth = [(fname, path)]
+        else:
+            fnames = os.listdir(path)
+            if not overwrite:
```
[the remark this answers is not in the document]
Comment (rlizzo) [reply src/hangar/cli/cli.py]: fixed!

File: src/hangar/cli/io/manage_plugins.py [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        the plugin is set for all functions.
+
+    See Also
+    --------
+    available_plugins : List of available plugins
+
+    Examples
+    --------
+
+    To use Matplotlib as the default image reader, you would write:
```
[the remark this answers is not in the document]
Comment (rlizzo) [reply src/hangar/cli/io/manage_plugins.py]: good catch! Fixed.

File: src/hangar/cli/io/manage_plugins.py (hunk 1 above)
Comment (rlizzo) [reply src/hangar/cli/io/manage_plugins.py]: Let's leave it. This is going to need some work in the future anyways.

**Comment by rlizzo** [comment]
Ok. Going to merge this now.

**Pull request merged**